extensions/wikia/ARecoveryEngine/js/SourcePoint/mms_client_fallback.js

```diff
@@ -0,0 +1,141 @@
+(function() {
+
+    window._sp_ = window._sp_ || {};
+   
+    // Init MMS client cmd queue
+    window._sp_.mms = window._sp_.mms || {};
+    window._sp_.mms.cmd = window._sp_.mms.cmd || [];
+    window._sp_.mms._internal = window._sp_.mms._internal || {};
+
+    // Init MMS API
+    var targetingParams = {};
+    window._sp_.mms.setTargeting = function(key, value) {
+        targetingParams[key] = value;
+    };
+
+    window._sp_.mms.startMsg = function() {
+        window._sp_.messagingDetection(loadMessage);
+    };
+
+    window._sp_.mms._internal.cdc1 = function() {
+        if (typeof window._sp_.config.mms_client_data_callback === 'function') {
+            try {
+                window._sp_.config.mms_client_data_callback.apply(
+                    window._sp_.config.mms_client_data_callback,
+                    arguments
+                );(function() {
+
+                    window._sp_ = window._sp_ || {};
+
+                    // Init MMS client cmd queue
+                    window._sp_.mms = window._sp_.mms || {};
+                    window._sp_.mms.cmd = window._sp_.mms.cmd || [];
+                    window._sp_.mms._internal = window._sp_.mms._internal || {};
+
+                    // Init MMS API
+                    var targetingParams = {};
+                    window._sp_.mms.setTargeting = function(key, value) {
+                        targetingParams[key] = value;
+                    };
+
+                    window._sp_.mms.startMsg = function() {
+                        window._sp_.messagingDetection(loadMessage);
+                    };
+
+                    window._sp_.mms._internal.cdc1 = function() {
+                        if (typeof window._sp_.config.mms_client_data_callback === 'function') {
+                            try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ANestedIfDepth)